pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1060,6 +1031,72 @@ func applyUpstreamTLSSettings(opts *buildClusterOpts, tls *networking.ClientTLSS
 				tlsContext.CommonTlsContext.AlpnProtocols = util.ALPNInMesh
 			}
 		}
+
+	case networking.ClientTLSSettings_SIMPLE, networking.ClientTLSSettings_MUTUAL:
+		if tls.Mode == networking.ClientTLSSettings_SIMPLE {
+			tlsContext = &auth.UpstreamTlsContext{
+				CommonTlsContext: &auth.CommonTlsContext{
+					ValidationContextType: &auth.CommonTlsContext_ValidationContext{
+						ValidationContext: certValidationContext,
+					},
+				},
+				Sni: tls.Sni,
+			}
+		} else if tls.Mode == networking.ClientTLSSettings_MUTUAL {
+			if tls.ClientCertificate == "" || tls.PrivateKey == "" {
+				log.Errorf("failed to apply tls setting for %s: client certificate and private key must not be empty",
+					c.Name)
+				return
+			}
+			tlsContext = &auth.UpstreamTlsContext{
+				CommonTlsContext: &auth.CommonTlsContext{},
+				Sni:              tls.Sni,
+			}
+
+			// Fallback to file mount secret instead of SDS if meshConfig.sdsUdsPath isn't set or SDS is disabled
```

Comment:
shouldn't this code be outside the elseif, as its common to simple and mutual?

Reply:
Looking at master, I don't see this being used for SIMPLE mode. But only for ISTIO_MUTUAL and MUTUAL

Reply:
This is also confusing, why are we not using file based sds for SIMPLE mode's root cert @howardjohn